pkgs/common-updater/unstable-updater.nix

```diff
@@ -0,0 +1,34 @@
+{ stdenv, writeScript, coreutils, git, common-updater-scripts }:
+
+# This is an updater for unstable packages that should always use the latest
+# commit.
+{ attrPath
+, url # The git url, usually just src.url
+}:
+
+let
+  updateScript = writeScript "unstable-update-script.sh" ''
+    #! ${stdenv.shell}
+    set -ex
+
+    attr_path="$1"
+    url="$2"
+
+    # Get info about HEAD from a shallow git clone
+    tmpdir="$(${coreutils}/bin/mktemp -d)"
+    ${git}/bin/git clone --bare --depth=1 "$url" "$tmpdir"
+    pushd "$tmpdir"
+    commit_date="$(${git}/bin/git show -s --pretty='format:%cs')"
+    commit_sha="$(${git}/bin/git show -s --pretty='format:%H')"
+    popd
+    ${coreutils}/bin/rm -rf "$tmpdir"
+
+    # update the nix expression
+    ${common-updater-scripts}/bin/update-source-version "$attr_path" \
+      "unstable-$commit_date" \
```

Comment:
Relevant: https://github.com/NixOS/nixpkgs/pull/100833

Reply:
Not sure whether to leave it as is until #100833 is in, or to make it configurable by taking a `makeVersionString` function as an argument (with default `date → "unstable-$date"`)

Reply:
I think I'll leave it as is for now